PluginGenerator/PluginBuilder.cs

```diff
@@ -62,19 +68,24 @@ public class PluginBuilder
 
         #region Public methods
 
-        public PluginBuilder(IJdkWrapper jdkWrapper, ILogger logger)
+        public PluginBuilder(IJdkWrapper jdkWrapper, IMavenArtifactHandler artifactHandler, ILogger logger)
```

Comment:
Previously the builders were extracting the jars from embedded resources. Now they ask the supplied artifact handler to fetch the required jars. This gives us a "seam" we can use when testing to inject a mock implementation.